### DISCUSSION TITLE
Customize XDG Desktop entry

### DISCUSSION THREAD

**Discussion by Petemir** [filips123/PWAsForFirefox#494] [Q&A] [answered]
Hello!

I  recently started using PWAsForFirefox and liking it so far :), thank you all for your work. Previously I manually created shortcuts and Desktop entries for individual apps/sites on different Chrome profiles on single-app mode, so I am happy that there is a similar (and automatic!) solution for Firefox.

As I understand that tray icons are difficult to deal with on multiplatform apps (specially on Linux!), as noted on #146, I normally use kdocker [0]. This means that for each new app I add to my system, I need to wrap around its call with a new script.

Although I could create new scripts and Desktop files for PWAs-with-kdocker, I would then end up with duplicated entries for each app. Of course I could delete the original ones, but I am uncertain of the consequences w.r.t. to how PWAsForFirefox admins the apps. 

Another way is manually modifying the `Exec` key from the current Desktop entries (what I am currently doing), but I fear that my changes may get modified whenever PWAsForFirefox makes changes/updates to any of the apps.

Therefore, I would like to know if there is any possibility to customize how the `.desktop` files get created, so I can change the Exec key for a call to my kdocker script.

Thanks!

[0] A bit outdated/unmaintained, but still works mostly OK; particularly for PWAs, I found the snippet at user-none/Kdocker#121 to be useful .

**Comment by filips123** [comment, marked as answer]
I think there is currently no reliable way to do this. You can remove or manually modify the generated Desktop Entries, but they will get regenerated/overwritten when you update/edit the web app (for example, from the extension). If you won't be updating web apps often, I guess that you can try this.

Or, maybe there is some way that you can hide original desktop entries without removing or editing them?

I'm not sure how this could be properly implemented, as it's quite specific functionality...

**Reply by Petemir** [reply]
Yes, I totally understand! I am always on the edge cases with my use :) . 

My first idea was to make a script to edit the Exec line in the generated Desktop Entries. I just wanted first to double-check if there was another, possibly less bug-prone, path. I guess I'll do that then :).

Thanks!
